### PR TITLE
search frontend: copy path instead of link for button displayed next to paths

### PR DESCRIPTION
**Pull request by rvantonder**
This PR changes the behavior of the copy file displayed next to paths, to copy paths, and not generate a link. This PR is a proposal looking for input, but can also be approved if there's enough consensus.

Screen Shot 2021-07-29 at 4 43 36 PM

We _can_ AB-test this change, but I don't want to propose an AB test for something that might not fly from the get-go from a design perspective. If there is enough consensus that this is a possible improvement and we'd like to AB test it first, then I'm willing to make appropriate changes. From a design perspective:

(1) It seems duplicative to have a permalink, and this copy button that generates a link. Maybe one solution is to move the permalink button next to the clipboard button, so both options are next to each other.

(2) It's annoying that there isn't a way to click to copy the path. Highlighting the regex path in the query bar or URL just doesn't stack up to a clipboard-button-copy. Dragging over the path doesn't work on all browsers and it also doesn't stack up to the convenience of a button click.

---

### Historical discussions / user issues

- "I miss being able to copy the file path and paste it into vim [...]" - customer user. [slack](https://sourcegraph.slack.com/archives/C0W2E592M/p1617404928119600?thread_ts=1617397296.117600&cid=C0W2E592M) Responses (I don't think I'm taking it out of context, please correct me if I'm misinterpreting the thread):
  - "It makes sense to me" - Alicja [slack](https://sourcegraph.slack.com/archives/C0HMGV90V/p1617796038079700)
  - "Makes perfect sense to me! - "Quinn Keast" (same thread as above)

- "Maybe the copy button should copy the path and not the link?" - Juliana [slack](https://sourcegraph.slack.com/archives/C0W2E592M/p1617405066120400?thread_ts=1617397296.117600&cid=C0W2E592M)

- "Speaking of the “copy link to clipboard” button. I was surprised that it copied the URL instead of the file path. There’s no good way to copy the relative filename while I suspect most people copy the URL from the URL bar instead of using “copy link to clipboard”. - Olaf [slack](https://sourcegraph.slack.com/archives/C0W2E592M/p1614936729113200)
- "Some people are asking about how to copy-paste the relative filename] - Olaf pre-Sourcegraph [slack](https://sourcegraph.slack.com/archives/C0W2E592M/p1584455913098000)

---

### Compared to GH

I'll point out how GH deals with some of their link copies:

File browser (explicit option to copy path or permalink):

Screen Shot 2021-07-29 at 4 48 56 PM

Pull requests (clicking the clipboard copies the _path_, this is actually really useful):

Screen Shot 2021-07-29 at 4 49 59 PM

Comparatively, our `-/commit/` view doesn't have corresponding way to copy paths:

Screen Shot 2021-07-29 at 4 52 22 PM

### CODE REVIEW

File: client/web/src/repo/actions/CopyPathAction.tsx [hunk 1]
```diff
@@ -25,10 +25,11 @@ export const CopyLinkAction: React.FunctionComponent = () => {
...
     const onClick = (event: React.MouseEvent<HTMLButtonElement>): void => {
         event.preventDefault()
-        eventLogger.log('ShareButtonClicked')
-        const shareLink = new URL(location.pathname + location.search + location.hash, window.location.href)
-        shareLink.searchParams.set('utm_source', 'share')
-        copy(shareLink.href)
+        eventLogger.log('CopyFilePath')
+        console.log(`Input: ${location.pathname}`)
+        // strip prefixes for URL paths matching `/-/blob/` (filepaths), `/-/tree/` (directory paths), and leading `/` (repository paths).
+        const path = location.pathname.replace(/(^.*?\/-\/(blob|tree)\/)|(^\/)/, '')
```
Comment (rvantonder) [inline client/web/src/repo/actions/CopyPathAction.tsx:31]: Don't know if using a regex here is too crazy. Seems OK: I don't know if there's a better source of truth for the path corresponding to a view. If there is I can use that instead.

Comment (olafurpg) [reply client/web/src/repo/actions/CopyPathAction.tsx:31]: There is a `parseBrowserRepoURL` helper method you can use https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/client/web/src/util/url.ts?L66:17#tab=references

**Review by olafurpg — APPROVED** [review]
Thank you for this PR! Big 👍 for changing this button to copy the relative file path.

File: client/web/src/repo/actions/CopyPathAction.tsx [hunk 2]
```diff
@@ -25,10 +25,11 @@ export const CopyLinkAction: React.FunctionComponent = () => {
 
     const onClick = (event: React.MouseEvent<HTMLButtonElement>): void => {
         event.preventDefault()
-        eventLogger.log('ShareButtonClicked')
-        const shareLink = new URL(location.pathname + location.search + location.hash, window.location.href)
-        shareLink.searchParams.set('utm_source', 'share')
-        copy(shareLink.href)
+        eventLogger.log('CopyFilePath')
+        console.log(`Input: ${location.pathname}`)
```
Comment (limitedmage) [inline client/web/src/repo/actions/CopyPathAction.tsx:29]:
```suggestion
```

**Review by limitedmage — APPROVED** [review]

File: client/web/src/repo/actions/CopyPathAction.tsx (hunk 1 above)
Comment (rvantonder) [reply client/web/src/repo/actions/CopyPathAction.tsx:31]: sweet!

**Review by rrhyne — APPROVED** [review]
Makes sense, thanks for making this change. FYI, I do not think this kind of change is a good candidate for an A/B test. It is hard to measure success when users take this data to another application.

**Comment by rvantonder** [comment]
Thanks @rrhyne makes sense!

**Pull request merged**